doc/source/serve/production.md

```diff
@@ -567,3 +557,216 @@ This section summarizes the best practices when deploying to production:
 * Use `serve config` to check the latest config that your Serve application received. This is its goal state.
 * Make lightweight configuration updates (e.g. `num_replicas` or `user_config` changes) by modifying your Serve config file and redeploying it with `serve deploy`.
 * Make heavyweight code updates (e.g. `runtime_env` changes) by starting a new Ray cluster, updating your Serve config file, and deploying the file with `serve deploy` to the new cluster. Once the new deployment is finished, switch your traffic to the new cluster.
+
+(serve-in-production-kubernetes)=
+
+## Deploying to Production on Kubernetes
+
+The above sections walk through how to deploy a Serve application manually using the Serve CLI.
+However, the best practice when running on Kubernetes is to use the [RayService] custom resource (CR) that's provided as part of [KubeRay] because it automatically handles important production requirements such as health checking, status reporting, failure recovery, and upgrades.
```

Comment:
There should probably be a broader sentence here mentioning that Ray Serve _can_ be run on Kubernetes. Non-Ray users that are using Kubernetes might not realize that's possible. The intro could probably discuss

1. Ray Serve can be run on K8s
2. Why should/shouldn't users run on K8s
3. (Optionally) why use Ray Serve at all/why not just use K8s

Reply:
done